Welcome to our repository to prepare for our upcoming workshop *Mathematical Frameworks for Integrative Analysis of Emerging Biological Data Type* **June 14- 19, 2020 in Banff, Canada**.

More details about the workshop can be found in [here](https://www.birs.ca/events/2020/5-day-workshops/20w5197) from the BIRS event page.

# Organisers
- [Aedin Culhane](https://www.hsph.harvard.edu/aedin-culhane/) (Harvard TH Chan School of Public Health)
- [Elana Fertig](https://fertiglab.com/) (John Hopkins)
- [Kim-Anh LÃª Cao](https://lecao-lab.science.unimelb.edu.au/) (University of Melbourne)

# Multi-omics studies
*Description of three multi-omics studies will be posted here, with easy access*.

## Gastrulation scNMT-seq data 
Three omics and more on 826 matching single cells, see the [Data folder scNMT-seq](https://github.com/BIRSBiointegration/Hackathon/tree/master/scNMT-seq) that contains a README and guidelines (including a list of challenges to tackle, plus more if you wish to help out).

## Cortex seq-FISH data
Two omics with ~1600 single cells each and 113 matched genes, see the [Data folder seqFISH](https://github.com/BIRSBiointegration/Hackathon/tree/master/seqFISH) that contains a README and challenges to address. Links to additional data are included, if you want to explore further. 

## Single-cell proteomic data
Two targeted proteomic studies investigating the tumor-immune microenvironment in breast cancer patients using different technologies (mass cytometry and MIBI-TOF), see the [Data folder sc-targeted-proteomics](https://github.com/BIRSBiointegration/Hackathon/tree/master/sc-targeted-proteomics) which contains a README and more details on the data, with some questions to begin exploring.

# Communication
All communication will be via Slack. Join our channel at birsbiointegration.slack.com at this [link](https://join.slack.com/t/birsbiointegration/shared_invite/enQtNzc4Njk4OTMwNDAzLTk2MTU2ZGQyMWE5ZTc1MmU3NDU2ZmFiNzRmMzcwYjZkN2JkYjcyNTU5NGJhNzhiNWQ4NjI3NmNjYmQxZWZkNGU) or contact aedin[@]jimmy.harvard.edu

# Timelines
- Studies start to be released from the week of **14th October 2019**.
- Abstract due on **March 1st 2020** (more details to come on the format - we expect preliminary analyses to be presented) 
- Abstract selection announced **March 14th 2020** 
- Reproducible analyses shared on our GitHub on **June 1st 2020**

